pkg/dao/dao.go

```diff
@@ -27,7 +23,7 @@ type Dao struct {
 	config    Config
```

Comment:
> Thinking about it more, the app.go code should delegate the validation of a data connection to the dao and the dao determines what that means since it really belongs in the domain of that package. A good indicator of that abstraction leak is if other code in the broker is importing etcd/client.

HEH I had a similar thought. Though I know like @shawn-hurley said the broker now imports the `etcd/client` simply to verify the errors that come back from it. I think it might be a good idea to define errors the broker knows about and keep the low level errors down in their own layers. i.e. broker shouldn't get an etcd error? but that would require some nasty handling since there are no Exceptions :)